FirebaseAppCheck/Sources/AppAttestProvider/FIRAppAttestProvider.m

```diff
@@ -207,6 +208,31 @@ - (void)getTokenWithCompletion:(void (^)(FIRAppCheckToken *_Nullable, NSError *_
       });
 }
 
+- (FBLPromise<FIRAppAttestKeyAttestationResult *> *)attestKey:(NSString *)keyID
+                                                    challenge:(NSData *)challenge {
+  return [FBLPromise onQueue:self.queue
+                          do:^NSData *_Nullable {
+                            return [FIRAppCheckCryptoUtils sha256HashFromData:challenge];
+                          }]
+      .thenOn(
+          self.queue,
+          ^FBLPromise<NSData *> *(NSData *challengeHash) {
+            return [FBLPromise onQueue:self.queue
+                wrapObjectOrErrorCompletion:^(FBLPromiseObjectOrErrorCompletion _Nonnull handler) {
+                  [self.appAttestService attestKey:keyID
+                                    clientDataHash:challengeHash
+                                 completionHandler:handler];
+                }];
+          })
```

Comment:
I'm seeing that we use `.thenOn(queue, ...)` and also return a `[FBLPromise onQueue:queue ... ]`

Question about Promises library:  is specifying the target queue required in both locations for cases like this?

Reply:
Good question. If I understand it correctly, yes, it's required for both as they actually may target different queues.